cms/djangoapps/contentstore/management/commands/tests/test_migrate_transcripts.py

```diff
@@ -0,0 +1,176 @@
+    """
+    def setUp(self):
+        super(TestArgParsing, self).setUp()
+
+    def test_no_args(self):
+        errstring = "At least one course or --all-courses must be specified."
+        with self.assertRaisesRegexp(CommandError, errstring):
+            call_command('migrate_transcripts')
+
+    def test_invalid_course(self):
+        with self.assertRaises(CommandError):
+            call_command('migrate_transcripts', "invalid-course")
+
+
+class MigrateTranscripts(ModuleStoreTestCase):
+    """
+    Tests migrating video transcripts in courses from contentstore to S3
+    """
+    def setUp(self):
+        """ Common setup. """
+        super(MigrateTranscripts, self).setUp()
+
+        self.store = modulestore()._get_modulestore_by_type(ModuleStoreEnum.Type.mongo)
+
+        self.course = CourseFactory.create()
+
+        video = {
+            "edx_video_id": "test_edx_video_id",
+            "client_video_id": "test1.mp4",
+            "duration": 42.0,
+            "status": "upload",
+            "courses": [unicode(self.course.id)],
+            "encoded_videos": [],
+            "created": datetime.now(pytz.utc)
+        }
+        api.create_video(video)
+
+        video_sample_xml = '''
+            <video display_name="Test Video"
+                   edx_video_id="test_edx_video_id"
+                   youtube="1.0:p2Q6BrNhdh8,0.75:izygArpw-Qo,1.25:1EeWXzPdhSA,1.5:rABDYkeK0x8"
+                   show_captions="false"
+                   download_track="false"
+                   start_time="00:00:01"
+                   download_video="false"
+                   end_time="00:01:00">
+              <source src="http://www.example.com/source.mp4"/>
+              <track src="http://www.example.com/track"/>
+              <handout src="http://www.example.com/handout"/>
+              <transcript language="ge" src="subs_grmtran1.srt" />
+              <transcript language="hr" src="subs_croatian1.srt" />
+            </video>
+        '''
+        self.video_descriptor = ItemFactory.create(
+            parent_location=self.course.location, category='video',
+            data={'data': video_sample_xml}
+        )
+
+        save_to_store(SRT_FILEDATA, 'subs_grmtran1.srt', 'text/srt', self.video_descriptor.location)
+        save_to_store(CRO_SRT_FILEDATA, 'subs_croatian1.srt', 'text/srt', self.video_descriptor.location)
+
+    def test_migrated_transcripts_count(self):
+
+        # check that transcript does not exist
+        languages = api.get_available_transcript_languages(self.video_descriptor.edx_video_id)
+        self.assertEqual(len(languages), 0)
+
+        # now call migrate_transcripts command and check the transcript availability
+        call_command('migrate_transcripts', unicode(self.course.id), '--commit')
+
+        languages = api.get_available_transcript_languages(self.video_descriptor.edx_video_id)
+        self.assertEqual(len(languages), 2)
+
+    def test_migrated_transcripts_without_commit(self):
```

Comment:
same comment as above.

Reply:
The command output is checked by the following test
 def test_migrate_transcripts_logging